hibernate-core/src/main/java/org/hibernate/event/internal/AbstractSaveEventListener.java

```diff
@@ -320,42 +316,6 @@ protected Serializable performSaveOrReplicate(
 		return id;
 	}
 
-	private static boolean shouldDelayIdentityInserts(boolean requiresImmediateIdAccess, EventSource source, EntityPersister persister) {
-		return shouldDelayIdentityInserts( requiresImmediateIdAccess, isPartOfTransaction( source ), source.getHibernateFlushMode(), persister );
-	}
-
-	private static boolean shouldDelayIdentityInserts(
-			boolean requiresImmediateIdAccess,
-			boolean partOfTransaction,
-			FlushMode flushMode,
-			EntityPersister persister) {
-		if ( !persister.getFactory().getSessionFactoryOptions().isPostInsertIdentifierDelayableEnabled() ) {
-			return false;
-		}
-
-		if ( requiresImmediateIdAccess ) {
-			// todo : make this configurable?  as a way to support this behavior with Session#save etc
-			return false;
-		}
-
-		// otherwise we should delay the IDENTITY insertions if either:
-		//		1) we are not part of a transaction
-		//		2) we are in FlushMode MANUAL or COMMIT (not AUTO nor ALWAYS)
-		if ( !partOfTransaction || flushMode == MANUAL || flushMode == COMMIT ) {
-			if ( persister.canIdentityInsertBeDelayed() ) {
-				return true;
-			}
-			LOG.debugf(
-					"Identity insert for entity [%s] should be delayed; however the persister requested early insert.",
-					persister.getEntityName()
-			);
-			return false;
-		}
-		else {
-			return false;
-		}
-	}
-
 	private static boolean isPartOfTransaction(EventSource source) {
 		return source.isTransactionInProgress() && source.getTransactionCoordinator().isJoined();
```

Comment:
I looked in 5.1 and don't see the "isJoined" check, in the reverted commit, we are now checking if the session is joined to the active JTA transaction, which seems like a good idea to me.  I think that we are reverting to code changes that may not be as well tested on WildFly, as the current ORM 5.3.x branch (with the reported bugs) or 5.1.x.  My call to action here is that we definitely should run the WF tests and the Eclipse Jakarta EE CTS tests with the 5.3.x branch (either via a snapshot release or an actual release).

Reply:
@scottmarlow Just so I am clear, are you suggesting we at a minimum merge a PR into 5.3 that reverts this logic just to test?  If so, that'll be up to @gbadner.  Just bear in mind we're trying to manage keeping 3 concurrent branches synchronized with this revert and I would prefer to keep the process as clean and clear as possible.

Reply:
> My only question is whether the code that we are reverting back to, will need further changes, or if we already tested that with WildFly in the past.

@scottmarlow I think in order to get this back to be identical to what has been tested with WF in the past, there is one additional revert I overlooked that you pointed out - which was the `#isJoined()` check.  

I overlooked that `#isJoined()`  was added as a part of HHH-11019 and therefore by removing it as well with the other things this PR includes, would return ORM 5.3 and 5.4 as identical to 5.1. I am fine keeping it as-is because I don't believe it will introduce any negative impact on behavior.

I'll wait to hear Gail's thoughts and go from there.  We're trying to get 5.4.1 out probably sometime next week, so just trying to plan accordingly here.

Reply:
@Naros I have no idea if the #isJoined() check should be removed, as there can be an active JTA transaction but the source (Hibernate Session?) may not be joined to it.  I agree that 5.1.x doesn't seem to contain that check.  I also don't know if requiresImmediateIdAccess already reflects whether the source is joined to the active JTA transaction.  @gbadner any thoughts on this?

Reply:
I believe checking `source.getTransactionCoordinator().isJoined()` is redundant.

`AbstractSharedSessionContract#isTransactionInProgress` calls `transactionCoordinator.isTransactionActive()`, which only returns`true` if `isJoined() == true`.

Reply:
@gbadner I don't believe we have those on CI (yet) afaik.

Reply:
I still see the redundant `source.getTransactionCoordinator().isJoined()` check...

Reply:
@gbadner Fixed in the last resync of the PR.